packages/app/src/systems/Transaction/hooks/useRawPayload.tsx

```diff
@@ -0,0 +1,23 @@
+import { hexlify, TransactionCoder, type Transaction } from 'fuels';
+import { useMemo } from 'react';
+
+type RawPayloadProps = {
+  transaction?: Transaction;
+};
+
+export function useRawPayload(props: Partial<RawPayloadProps>) {
+  const { transaction } = props;
+  const rawPayload = useMemo(() => {
+    if (transaction) {
+      try {
+        const raw = hexlify(new TransactionCoder().encode(transaction));
+
+        return raw;
+        // eslint-disable-next-line no-empty
+      } catch (_) {}
```

Comment:
should we throw an error or do a toast message if there is an error?

Reply:
no, the transaction might be incomplete and we don't wanna show it to the user, we just need to ignore and avoid crash